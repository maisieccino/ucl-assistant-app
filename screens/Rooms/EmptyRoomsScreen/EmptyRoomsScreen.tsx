import { Picker } from '@react-native-community/picker'
import { StackNavigationProp } from '@react-navigation/stack'
import React from 'react'
import {
  ActivityIndicator,
  StyleSheet,
  View,
} from 'react-native'
import { connect, ConnectedProps } from "react-redux"
import { Page } from "../../../components/Containers"
import SearchResult from "../../../components/SearchResult"
import { BodyText, CentredText } from "../../../components/Typography"
import type { AppStateType } from '../../../configureStore'
import ApiManager from "../../../lib/ApiManager"
import type { Room } from '../../../types/uclapi'
import type { RoomsNavigatorParamList } from "../RoomsNavigator"

const styles = StyleSheet.create({
  container: {
    paddingBottom: 20,
  },
  subtitle: {
    marginBottom: 10,
  },
})

interface Props extends PropsFromRedux {
  navigation: StackNavigationProp<RoomsNavigatorParamList>,
}

interface State {
  emptyRooms: Array<Room>,
  error: Error,
  loadingEmptyRooms: boolean,
  selectedSite: string,
  sites: Array<unknown>,
}

export class EmptyRoomsScreen extends React.Component<
  Props,
  State
  > {
  static navigationOptions = {
    title: `Empty Rooms`,
  }

  constructor(props: Props) {
    super(props)
    this.state = {
      emptyRooms: [],
      error: null,
      loadingEmptyRooms: true,
      selectedSite: null,
      sites: [],
    }
  }

  componentDidMount(): void {
    this.fetchEmptyRooms()
  }

  fetchEmptyRooms = async (): Promise<void> => {
    const { token } = this.props
    try {
      const emptyRooms = await ApiManager.rooms.getEmptyRooms(token)
      const sites = Array.from(new Set(emptyRooms.map(({ siteid }) => siteid)))
        .map((siteid) => ({
          siteid,
          sitename: emptyRooms.find((r) => r.siteid === siteid).sitename,
        }))
      this.setState({
        emptyRooms,
        loadingEmptyRooms: false,
        sites,
      })
    } catch (error) {
      this.setState({ error: error.message })
    }
  }

  navigateToRoomDetail = (room: Room) => (): void => {
    const { navigation } = this.props
    navigation.navigate(`RoomsDetail`, { room })
  }

  onSelectSite = (value: string): void => {
    this.setState({ selectedSite: value })
  }

  renderEmptyRoom = (room = null): React.ReactNode => {
    if (room === null) {
      return null
    }

    const key = `${room.roomid}-${room.roomname}-`
      + `${room.siteid}-${room.classification_name}`

    return (
      <SearchResult
        key={key}
        topText={room.roomname}
        bottomText={room.classification_name}
        type="location"
        onPress={this.navigateToRoomDetail(room)}
      />
    )
  }

  renderResults = (): React.ReactNode => {
    const { emptyRooms, selectedSite } = this.state
    const matchingRooms = emptyRooms.filter(({ siteid }) => {
      if (selectedSite === null) {
        return true
      }
      return siteid === selectedSite
    })

    if (matchingRooms.length === 0) {
      return (
        <View>
          <CentredText>
            No empty rooms found üôÅ
          </CentredText>
        </View>

      )
    }
    return matchingRooms.map(this.renderEmptyRoom)
  }

  render(): React.ReactElement {
    const {
      error,
      sites,
      selectedSite,
      loadingEmptyRooms,
    } = this.state

    if (error) {
      return <BodyText>{`Could not fetch empty rooms: ${error}`}</BodyText>
    }
    if (loadingEmptyRooms) {
      return (
        <Page>
          <View style={styles.container}>
            <ActivityIndicator size="large" />
          </View>
        </Page>
      )
    }
    return (
      <Page>
        <View style={styles.container}>
          <BodyText>
            Filter by:
          </BodyText>
          <Picker
            selectedValue={selectedSite}
            onValueChange={this.onSelectSite}
            testID="building-picker"
          >
            <Picker.Item label="All Buildings" value={null} />
            {
              sites.map(({ sitename, siteid }) => (
                <Picker.Item key={siteid} label={sitename} value={siteid} />
              ))
            }
          </Picker>
          <BodyText style={styles.subtitle}>
            Rooms vacant for the next hour:
          </BodyText>
          {this.renderResults()}
        </View>
      </Page>
    )
  }
}

const connector = connect(
  (state: AppStateType) => ({
    token: state.user.token,
  }),
  () => ({}),
)

type PropsFromRedux = ConnectedProps<typeof connector>

export default connector(EmptyRoomsScreen)
