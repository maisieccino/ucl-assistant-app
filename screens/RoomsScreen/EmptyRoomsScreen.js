import PropTypes from "prop-types"
import React from 'react'
import {
  ActivityIndicator,
  Picker,
  StyleSheet,
  View,
} from 'react-native'
import { connect } from "react-redux"

import { Page } from "../../components/Containers"
import SearchResult from "../../components/SearchResult"
import { BodyText, CentredText } from "../../components/Typography"
import ApiManager from "../../lib/ApiManager"

const styles = StyleSheet.create({
  container: {
    paddingBottom: 20,
  },
  subtitle: {
    marginBottom: 10,
  },
})

export class EmptyRoomsScreen extends React.Component {
  static navigationOptions = {
    title: `Empty Rooms`,
  }

  static mapStateToProps = (state) => ({
    token: state.user.token,
  })

  static propTypes = {
    navigation: PropTypes.shape().isRequired,
    token: PropTypes.string,
  }

  static defaultProps = {
    token: ``,
  }

  constructor() {
    super()
    this.state = {
      emptyRooms: [],
      error: null,
      loadingEmptyRooms: true,
      selectedSite: null,
      sites: [],
    }
  }

  componentDidMount() {
    this.fetchEmptyRooms()
  }

  fetchEmptyRooms = async () => {
    const { token } = this.props
    try {
      const emptyRooms = await ApiManager.rooms.getEmptyRooms(token)
      const sites = Array.from(new Set(emptyRooms.map(({ siteid }) => siteid)))
        .map((siteid) => ({
          siteid,
          sitename: emptyRooms.find((r) => r.siteid === siteid).sitename,
        }))
      this.setState({
        emptyRooms,
        loadingEmptyRooms: false,
        sites,
      })
    } catch (error) {
      this.setState({ error: error.message })
    }
  }

  navigateToRoomDetail = (room) => () => {
    const { navigation } = this.props
    navigation.navigate(`RoomDetail`, { room })
  }

  onSelectSite = (value) => {
    this.setState({ selectedSite: value })
  }

  renderEmptyRoom = (room = null) => {
    if (room === null) {
      return null
    }

    const key = `${room.roomid}-${room.roomname}-`
      + `${room.siteid}-${room.classification_name}`

    return (
      <SearchResult
        key={key}
        topText={room.roomname}
        bottomText={room.classification_name}
        type="location"
        buttonText="View"
        onPress={this.navigateToRoomDetail(room)}
      />
    )
  }

  renderResults = () => {
    const { emptyRooms, selectedSite } = this.state
    const matchingRooms = emptyRooms.filter(({ siteid }) => {
      if (selectedSite === null) {
        return true
      }
      return siteid === selectedSite
    })

    if (matchingRooms.length === 0) {
      return (
        <CentredText testID="empty-rooms-message">
          No empty rooms found :(
        </CentredText>
      )
    }
    return matchingRooms.map(this.renderEmptyRoom)
  }

  render() {
    const {
      error,
      sites,
      selectedSite,
      loadingEmptyRooms,
    } = this.state

    if (error) {
      return <BodyText>{`Could not fetch empty rooms: ${error}`}</BodyText>
    }
    if (loadingEmptyRooms) {
      return (
        <Page>
          <View style={styles.container}>
            <ActivityIndicator size="large" />
          </View>
        </Page>
      )
    }
    return (
      <Page>
        <View style={styles.container}>
          <BodyText>
            Filter by:
          </BodyText>
          <Picker
            selectedValue={selectedSite}
            onValueChange={this.onSelectSite}
            testID="building-picker"
          >
            <Picker.Item label="All Buildings" value={null} />
            {
              sites.map(({ sitename, siteid }) => (
                <Picker.Item key={siteid} label={sitename} value={siteid} />
              ))
            }
          </Picker>
          <BodyText style={styles.subtitle}>
            Rooms vacant for the next hour:
          </BodyText>
          {this.renderResults()}
        </View>
      </Page>
    )
  }
}

export default connect(
  EmptyRoomsScreen.mapStateToProps,
  () => ({}),
)(EmptyRoomsScreen)
