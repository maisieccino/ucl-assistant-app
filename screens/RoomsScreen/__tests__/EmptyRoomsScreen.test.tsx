/**
 * @jest-environment jsdom
 */
import "react-native"

import React from 'react'
import {
  cleanup, fireEvent, render, waitForElement,
} from 'react-native-testing-library'

import ApiManager from "../../../lib/ApiManager"
import { EmptyRoomsScreen } from '../EmptyRoomsScreen'

describe(`EmptyRoomsScreen`, () => {
  let wrapper
  const mockNavigate = jest.fn()
  const mockProps = {
    navigation: {
      navigate: mockNavigate,
    },
  }

  /* eslint-disable @typescript-eslint/camelcase */
  const sampleRooms = [
    {
      classification_name: `Classroom`,
      roomname: `Bentham`,
      siteid: `b`,
      sitename: `Bentham House`,
    },
    {
      classification_name: `Lecture Theatre`,
      roomname: `Galactica`,
      siteid: `main`,
      sitename: `Main Building`,
    },
    {
      classification_name: `Classroom`,
      roomname: `Denning`,
      siteid: `main`,
      sitename: `Main Building`,
    },
  ]
  /* eslint-enable @typescript-eslint/camelcase */

  const mockGetEmptyRooms = jest.fn(
    () => Promise.resolve(sampleRooms),
  ) as jest.Mock
  ApiManager.rooms.getEmptyRooms = mockGetEmptyRooms

  beforeEach(() => {
    jest.clearAllMocks()
    wrapper = render(<EmptyRoomsScreen {...mockProps} />)
  })

  afterEach(() => {
    cleanup()
  })

  it(`renders the loading screen`, async () => {
    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  it(`calls getEmptyRooms`, async () => {
    const { update } = wrapper
    update(<EmptyRoomsScreen {...mockProps} />)
    await new Promise((resolve) => setTimeout(resolve, 0))
    expect(mockGetEmptyRooms).toHaveBeenCalledTimes(1)
    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  it(`shows message when no empty rooms found`, async () => {
    ApiManager.rooms.getEmptyRooms = jest.fn(() => Promise.resolve([]))
    const emptyScreen = render(<EmptyRoomsScreen {...mockProps} />)
    const { update, getByTestId } = emptyScreen

    update(<EmptyRoomsScreen {...mockProps} />)
    await new Promise((resolve) => setTimeout(resolve, 0))

    await waitForElement(() => getByTestId(`empty-rooms-message`))
    expect(emptyScreen.toJSON()).toMatchSnapshot()

    ApiManager.rooms.getEmptyRooms = mockGetEmptyRooms
  })

  it(`supports filtering by building`, async () => {
    const { update, getByTestId } = wrapper
    update(<EmptyRoomsScreen {...mockProps} />)
    await new Promise((resolve) => setTimeout(resolve, 0))

    fireEvent(getByTestId(`building-picker`), `onValueChange`, `main`)
    update(<EmptyRoomsScreen {...mockProps} />)
    await new Promise((resolve) => setTimeout(resolve, 0))

    expect(wrapper.toJSON()).toMatchSnapshot()
  })
})
