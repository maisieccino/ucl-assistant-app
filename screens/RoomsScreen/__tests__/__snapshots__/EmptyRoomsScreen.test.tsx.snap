// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyRoomsScreen calls getEmptyRooms 1`] = `
Array [
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "flex": 0,
        },
        Object {
          "backgroundColor": "#fefefe",
        },
      ]
    }
  />,
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "backgroundColor": "#fefefe",
          "height": 1334,
          "width": 750,
        },
        Object {},
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "#fefefe",
              "height": 1334,
              "width": 750,
            },
            Object {},
          ],
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <RCTScrollView
        contentContainerStyle={
          Array [
            Object {
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 10,
            },
            Object {},
          ]
        }
        keyboardDismissMode="on-drag"
        refreshControl={null}
        style={
          Array [
            Object {
              "backgroundColor": "#fefefe",
              "flex": 1,
              "paddingLeft": 20,
              "paddingRight": 20,
            },
            Object {
              "paddingLeft": 0,
              "paddingRight": 0,
            },
            Object {},
          ]
        }
      >
        <View>
          <View
            style={
              Object {
                "height": 10,
              }
            }
          />
          <View
            style={
              Object {
                "paddingBottom": 20,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#4D5061",
                    "fontFamily": "apercu",
                    "fontSize": 16,
                    "marginBottom": 1,
                    "marginTop": 1,
                  },
                  Object {},
                ]
              }
            >
              Filter by:
            </Text>
            <View>
              <RCTPicker
                items={
                  Array [
                    Object {
                      "label": "All Buildings",
                      "textColor": undefined,
                      "value": null,
                    },
                    Object {
                      "label": "Bentham House",
                      "textColor": undefined,
                      "value": "b",
                    },
                    Object {
                      "label": "Main Building",
                      "textColor": undefined,
                      "value": "main",
                    },
                  ]
                }
                onChange={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                selectedIndex={0}
                style={
                  Array [
                    Object {
                      "height": 216,
                    },
                    undefined,
                  ]
                }
                testID="building-picker"
              />
            </View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#4D5061",
                    "fontFamily": "apercu",
                    "fontSize": 16,
                    "marginBottom": 1,
                    "marginTop": 1,
                  },
                  Object {
                    "marginBottom": 10,
                  },
                ]
              }
            >
              Rooms vacant for the next hour:
            </Text>
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    Array [
                      Object {
                        "alignSelf": "stretch",
                        "backgroundColor": "#fff",
                        "borderRadius": 10,
                        "marginBottom": 10,
                        "marginLeft": 2,
                        "marginRight": 2,
                        "marginTop": 5,
                        "padding": 10,
                        "shadowColor": "#000",
                        "shadowOffset": Object {
                          "height": 2,
                        },
                        "shadowOpacity": 0.1845,
                        "shadowRadius": 1.62,
                      },
                      Object {},
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#eaeaea",
                      "borderRadius": 80,
                      "marginRight": 10,
                      "padding": 10,
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": undefined,
                          "fontSize": 24,
                        },
                        undefined,
                        Object {
                          "fontFamily": "feather",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    ï†Ž
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#4D5061",
                          "fontFamily": "apercu-bold",
                          "fontSize": 16,
                          "marginBottom": 0,
                          "marginTop": 1,
                        },
                        Object {},
                      ]
                    }
                  >
                    Bentham
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#6E707F",
                          "fontFamily": "apercu",
                          "fontSize": 16,
                          "fontStyle": "italic",
                          "marginBottom": 1,
                          "marginTop": 0,
                        },
                        Object {},
                      ]
                    }
                  >
                    Classroom
                  </Text>
                </View>
              </View>
            </View>
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    Array [
                      Object {
                        "alignSelf": "stretch",
                        "backgroundColor": "#fff",
                        "borderRadius": 10,
                        "marginBottom": 10,
                        "marginLeft": 2,
                        "marginRight": 2,
                        "marginTop": 5,
                        "padding": 10,
                        "shadowColor": "#000",
                        "shadowOffset": Object {
                          "height": 2,
                        },
                        "shadowOpacity": 0.1845,
                        "shadowRadius": 1.62,
                      },
                      Object {},
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#eaeaea",
                      "borderRadius": 80,
                      "marginRight": 10,
                      "padding": 10,
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": undefined,
                          "fontSize": 24,
                        },
                        undefined,
                        Object {
                          "fontFamily": "feather",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    ï†Ž
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#4D5061",
                          "fontFamily": "apercu-bold",
                          "fontSize": 16,
                          "marginBottom": 0,
                          "marginTop": 1,
                        },
                        Object {},
                      ]
                    }
                  >
                    Galactica
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#6E707F",
                          "fontFamily": "apercu",
                          "fontSize": 16,
                          "fontStyle": "italic",
                          "marginBottom": 1,
                          "marginTop": 0,
                        },
                        Object {},
                      ]
                    }
                  >
                    Lecture Theatre
                  </Text>
                </View>
              </View>
            </View>
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    Array [
                      Object {
                        "alignSelf": "stretch",
                        "backgroundColor": "#fff",
                        "borderRadius": 10,
                        "marginBottom": 10,
                        "marginLeft": 2,
                        "marginRight": 2,
                        "marginTop": 5,
                        "padding": 10,
                        "shadowColor": "#000",
                        "shadowOffset": Object {
                          "height": 2,
                        },
                        "shadowOpacity": 0.1845,
                        "shadowRadius": 1.62,
                      },
                      Object {},
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#eaeaea",
                      "borderRadius": 80,
                      "marginRight": 10,
                      "padding": 10,
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": undefined,
                          "fontSize": 24,
                        },
                        undefined,
                        Object {
                          "fontFamily": "feather",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    ï†Ž
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#4D5061",
                          "fontFamily": "apercu-bold",
                          "fontSize": 16,
                          "marginBottom": 0,
                          "marginTop": 1,
                        },
                        Object {},
                      ]
                    }
                  >
                    Denning
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#6E707F",
                          "fontFamily": "apercu",
                          "fontSize": 16,
                          "fontStyle": "italic",
                          "marginBottom": 1,
                          "marginTop": 0,
                        },
                        Object {},
                      ]
                    }
                  >
                    Classroom
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTSafeAreaView>,
]
`;

exports[`EmptyRoomsScreen renders the loading screen 1`] = `
Array [
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "flex": 0,
        },
        Object {
          "backgroundColor": "#fefefe",
        },
      ]
    }
  />,
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "backgroundColor": "#fefefe",
          "height": 1334,
          "width": 750,
        },
        Object {},
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "#fefefe",
              "height": 1334,
              "width": 750,
            },
            Object {},
          ],
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <RCTScrollView
        contentContainerStyle={
          Array [
            Object {
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 10,
            },
            Object {},
          ]
        }
        keyboardDismissMode="on-drag"
        refreshControl={null}
        style={
          Array [
            Object {
              "backgroundColor": "#fefefe",
              "flex": 1,
              "paddingLeft": 20,
              "paddingRight": 20,
            },
            Object {
              "paddingLeft": 0,
              "paddingRight": 0,
            },
            Object {},
          ]
        }
      >
        <View>
          <View
            style={
              Object {
                "height": 10,
              }
            }
          />
          <View
            style={
              Object {
                "paddingBottom": 20,
              }
            }
          >
            <ActivityIndicator
              animating={true}
              color="#999999"
              hidesWhenStopped={true}
              size="large"
            />
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTSafeAreaView>,
]
`;

exports[`EmptyRoomsScreen shows message when no empty rooms found 1`] = `
Array [
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "flex": 0,
        },
        Object {
          "backgroundColor": "#fefefe",
        },
      ]
    }
  />,
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "backgroundColor": "#fefefe",
          "height": 1334,
          "width": 750,
        },
        Object {},
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "#fefefe",
              "height": 1334,
              "width": 750,
            },
            Object {},
          ],
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <RCTScrollView
        contentContainerStyle={
          Array [
            Object {
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 10,
            },
            Object {},
          ]
        }
        keyboardDismissMode="on-drag"
        refreshControl={null}
        style={
          Array [
            Object {
              "backgroundColor": "#fefefe",
              "flex": 1,
              "paddingLeft": 20,
              "paddingRight": 20,
            },
            Object {
              "paddingLeft": 0,
              "paddingRight": 0,
            },
            Object {},
          ]
        }
      >
        <View>
          <View
            style={
              Object {
                "height": 10,
              }
            }
          />
          <View
            style={
              Object {
                "paddingBottom": 20,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#4D5061",
                    "fontFamily": "apercu",
                    "fontSize": 16,
                    "marginBottom": 1,
                    "marginTop": 1,
                  },
                  Object {},
                ]
              }
            >
              Filter by:
            </Text>
            <View>
              <RCTPicker
                items={
                  Array [
                    Object {
                      "label": "All Buildings",
                      "textColor": undefined,
                      "value": null,
                    },
                  ]
                }
                onChange={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                selectedIndex={0}
                style={
                  Array [
                    Object {
                      "height": 216,
                    },
                    undefined,
                  ]
                }
                testID="building-picker"
              />
            </View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#4D5061",
                    "fontFamily": "apercu",
                    "fontSize": 16,
                    "marginBottom": 1,
                    "marginTop": 1,
                  },
                  Object {
                    "marginBottom": 10,
                  },
                ]
              }
            >
              Rooms vacant for the next hour:
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#6E707F",
                    "fontFamily": "apercu",
                    "fontSize": 16,
                    "marginBottom": 5,
                    "marginTop": 5,
                    "textAlign": "center",
                  },
                  Object {},
                ]
              }
            >
              No empty rooms found :(
            </Text>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTSafeAreaView>,
]
`;

exports[`EmptyRoomsScreen supports filtering by building 1`] = `
Array [
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "flex": 0,
        },
        Object {
          "backgroundColor": "#fefefe",
        },
      ]
    }
  />,
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "backgroundColor": "#fefefe",
          "height": 1334,
          "width": 750,
        },
        Object {},
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "#fefefe",
              "height": 1334,
              "width": 750,
            },
            Object {},
          ],
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <RCTScrollView
        contentContainerStyle={
          Array [
            Object {
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 10,
            },
            Object {},
          ]
        }
        keyboardDismissMode="on-drag"
        refreshControl={null}
        style={
          Array [
            Object {
              "backgroundColor": "#fefefe",
              "flex": 1,
              "paddingLeft": 20,
              "paddingRight": 20,
            },
            Object {
              "paddingLeft": 0,
              "paddingRight": 0,
            },
            Object {},
          ]
        }
      >
        <View>
          <View
            style={
              Object {
                "height": 10,
              }
            }
          />
          <View
            style={
              Object {
                "paddingBottom": 20,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#4D5061",
                    "fontFamily": "apercu",
                    "fontSize": 16,
                    "marginBottom": 1,
                    "marginTop": 1,
                  },
                  Object {},
                ]
              }
            >
              Filter by:
            </Text>
            <View>
              <RCTPicker
                items={
                  Array [
                    Object {
                      "label": "All Buildings",
                      "textColor": undefined,
                      "value": null,
                    },
                    Object {
                      "label": "Bentham House",
                      "textColor": undefined,
                      "value": "b",
                    },
                    Object {
                      "label": "Main Building",
                      "textColor": undefined,
                      "value": "main",
                    },
                  ]
                }
                onChange={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                selectedIndex={2}
                style={
                  Array [
                    Object {
                      "height": 216,
                    },
                    undefined,
                  ]
                }
                testID="building-picker"
              />
            </View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#4D5061",
                    "fontFamily": "apercu",
                    "fontSize": 16,
                    "marginBottom": 1,
                    "marginTop": 1,
                  },
                  Object {
                    "marginBottom": 10,
                  },
                ]
              }
            >
              Rooms vacant for the next hour:
            </Text>
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    Array [
                      Object {
                        "alignSelf": "stretch",
                        "backgroundColor": "#fff",
                        "borderRadius": 10,
                        "marginBottom": 10,
                        "marginLeft": 2,
                        "marginRight": 2,
                        "marginTop": 5,
                        "padding": 10,
                        "shadowColor": "#000",
                        "shadowOffset": Object {
                          "height": 2,
                        },
                        "shadowOpacity": 0.1845,
                        "shadowRadius": 1.62,
                      },
                      Object {},
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#eaeaea",
                      "borderRadius": 80,
                      "marginRight": 10,
                      "padding": 10,
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": undefined,
                          "fontSize": 24,
                        },
                        undefined,
                        Object {
                          "fontFamily": "feather",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    ï†Ž
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#4D5061",
                          "fontFamily": "apercu-bold",
                          "fontSize": 16,
                          "marginBottom": 0,
                          "marginTop": 1,
                        },
                        Object {},
                      ]
                    }
                  >
                    Galactica
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#6E707F",
                          "fontFamily": "apercu",
                          "fontSize": 16,
                          "fontStyle": "italic",
                          "marginBottom": 1,
                          "marginTop": 0,
                        },
                        Object {},
                      ]
                    }
                  >
                    Lecture Theatre
                  </Text>
                </View>
              </View>
            </View>
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    Array [
                      Object {
                        "alignSelf": "stretch",
                        "backgroundColor": "#fff",
                        "borderRadius": 10,
                        "marginBottom": 10,
                        "marginLeft": 2,
                        "marginRight": 2,
                        "marginTop": 5,
                        "padding": 10,
                        "shadowColor": "#000",
                        "shadowOffset": Object {
                          "height": 2,
                        },
                        "shadowOpacity": 0.1845,
                        "shadowRadius": 1.62,
                      },
                      Object {},
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#eaeaea",
                      "borderRadius": 80,
                      "marginRight": 10,
                      "padding": 10,
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": undefined,
                          "fontSize": 24,
                        },
                        undefined,
                        Object {
                          "fontFamily": "feather",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    ï†Ž
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#4D5061",
                          "fontFamily": "apercu-bold",
                          "fontSize": 16,
                          "marginBottom": 0,
                          "marginTop": 1,
                        },
                        Object {},
                      ]
                    }
                  >
                    Denning
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#6E707F",
                          "fontFamily": "apercu",
                          "fontSize": 16,
                          "fontStyle": "italic",
                          "marginBottom": 1,
                          "marginTop": 0,
                        },
                        Object {},
                      ]
                    }
                  >
                    Classroom
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTSafeAreaView>,
]
`;
